### ISSUE TITLE
Throws when using `copyTpl` and image types

### ISSUE THREAD

**Issue by lukeed** [SBoudrias/mem-fs-editor#65]
> My scenario & examples occur within the context of `yeoman-generator`, but `mem-fs-editor` is mounted directly as `this.fs`

When **copying** files from `templatePath` to `destinationPath`, I need to explicitly ignore all `.jpg` files (and presumably gif & png too):

```js
this.fs.copyTpl([
	`${this.templatePath()}/**`,
	'!**/test.js',
	'!**/*.jpg'
], this.destinationPath(), {...});
```

If I don't, the following stack is thrown:

```
Error: Could not find matching close tag for "<%".
    node_modules/ejs/lib/ejs.js:560:19
    Object.Template.generateSource (node_modules/ejs/lib/ejs.js:545:15)
    Object.Template.compile (node_modules/ejs/lib/ejs.js:450:12)
    Object.compile (node_modules/ejs/lib/ejs.js:306:16)
    handleCache (node_modules/ejs/lib/ejs.js:169:18)
    Object.exports.render (node_modules/ejs/lib/ejs.js:332:10)
    copy.extend.process (node_modules/mem-fs-editor/lib/actions/copy-tpl.js:12:18)
    applyProcessingFunc (node_modules/mem-fs-editor/lib/actions/copy.js:13:16)
    EditionInterface.exports._copySingle (node_modules/mem-fs-editor/lib/actions/copy.js:51:16)
    EditionInterface.exports.copy (node_modules/mem-fs-editor/lib/actions/copy.js:23:17)
    EditionInterface.module.exports [as copyTpl] (node_modules/mem-fs-editor/lib/actions/copy-tpl.js:10:8)
    cp (app/index.js:81:13)
    app/index.js:92:4
```

I know I can successfully copy images to my destination using `copy()`, but it can only be done one file at a time && makes the copy-templates phase clunkier.

```js
this.fs.copy(this.templatePath('assets/foo.jpg'), this.destinationPath('assets/foo.jpg'));
this.fs.copy(this.templatePath('assets/bar.jpg'), this.destinationPath('assets/bar.jpg'));
this.fs.copy(this.templatePath('assets/baz.jpg'), this.destinationPath('assets/baz.jpg'));
```

#### TL;DR;

Can we filter & ignore extensions within `copyTpl()` so that images can be cloned along with everything else in one fell swoop?

**Comment by SBoudrias** [comment]
Looks like you found out how to pass a glob ignore pattern.

`copy` also support glob pattern, so you can do the same to copy over your images `**/*.jpg`

**Comment by lukeed** [comment]
Well, yes, but that's not relevant to my suggestion / issue.

As mentioned in the TLDR, `copyTpl` should be ignoring certain file types since `ejs` can't do anything with them to begin with.

**Comment by SBoudrias** [comment]
A default ignore list wouldn't be a predictable behavior for the library users. We offers base functionality, it's up to the user to deal with which file they want to copy.

**Comment by loopmode** [comment]
It feels wrong to have to do this manually. I'm already at `[this.templatePath('**/*'), '!**/*.gif', '!**/*.png', '!**/*.zip', '!**/*.woff'],` and there's more coming I guess.

There should be a mechanism that automatically copies binary files without attempting to do template interpolations. That should only be done on text-based files.

**Comment by loopmode** [comment]
In case somebody stumbles across this and just doesn't have the time to deal with it.. this is one way to get around it. You'll need to install the [`binary-extensions` ](https://www.npmjs.com/package/binary-extensions)package.

```
const binaryExtensions = require('binary-extensions');
...
  
    writing() {
        // binary files often throw errors when copied via copyTpl
        // see https://github.com/SBoudrias/mem-fs-editor/issues/65
        // so we exclude them from fs.copyTpl, then manually fs.copy them

        const binaryTemplateFiles = glob.sync(`**/*.{${binaryExtensions.join(',')}}`, {
            cwd: path.resolve(this.templatePath()),
        });
        const binaryIgnoreGlobs = new Set(
            // using Set to drop duplicates from the array
            binaryTemplateFiles.map(file => `!**/*${path.extname(file)}`),
        );

        this.fs.copyTpl(
            [this.templatePath('**/*'), ...binaryIgnoreGlobs],
            this.destinationPath(),
            this.props,
            {},
            { globOptions: { dot: true } },
        );

        binaryTemplateFiles.forEach(file => {
            this.fs.copy(this.templatePath(file), this.destinationPath(file));
        });
    }
```

**Comment by SBoudrias** [comment]
@loopmode feel free to send a PR. I'll be happy to help review and get this merged

**Comment by loopmode** [comment]
Not sure if/when I'll find the time, but sure! By the way, the `binary-extensions` solution - based on a whitelist - already failed me when I randomly tried out stuff. I had an .eps file and that's not in the list, so I already had to extend it. I guess we'd need a better detection method.
Or something where users could provide extensions to be handled separately using a config file. Yeoman is the main use case for this lib, right? Maybe the whitelisting for defaults plus `.yo-rc.json` support for special cases could work out..
